### DISCUSSION TITLE
if no paths are defined server allows any to publish / deny options?

### DISCUSSION THREAD

**Discussion by t3therdev** [bluenviron/mediamtx#445] [Questions]
## Which version are you using?

v0.15.5

## Which operating system are you using?

OS

- [ ] Linux
- [X] Windows
- [ ] macOS

## Describe the issue

If not paths are defined the server will allow any client to publish. We are trying to dynamically control the application and at some points there will be no paths. How can we deny clients when no paths are created? Is there a deny all mode?

## Describe how to replicate the issue

1. start the server
2. publish with a path
3. remove the path
4. you can now publish with any path

how can we block this and ONLY allow publishing when a path is defined?

**Comment by aler9** [comment]
Hello, the problem is that when you remove a path, the server detects that there are no paths defined, and uses the default configuration, which is

```yml
paths:
  all:
```

Probably a better behavior is possible, in the meanwhile you can define an additional path to prevent the issue:

```yml
paths:
  unused:
    publishUser: myuser
    publishPass: mypass
```

[a comment by an automated account is not shown]
